ui/components/MesheryCatalog.js

```diff
@@ -0,0 +1,1093 @@
+      name: 'DEPLOY_PATTERN',
+      error_msg: 'Failed to deploy design file',
+    },
+    UNDEPLOY_PATTERN: {
+      name: 'UNDEPLOY_PATTERN',
+      error_msg: 'Failed to undeploy design file',
+    },
+    CLONE_PATTERN: {
+      name: 'CLONE_PATTERN',
+      error_msg: 'Failed to clone design file',
+    },
+    UNPUBLISH_CATALOG: {
+      name: 'PUBLISH_CATALOG',
+      error_msg: 'Failed to publish catalog',
+    },
+    SCHEMA_FETCH: {
+      name: 'SCHEMA_FETCH',
+      error_msg: 'failed to fetch import schema',
+    },
+  };
+
+  /**
+   * Checking whether users are signed in under a provider that doesn't have
+   * publish pattern capability and setting the canPublishPattern state accordingly
+   */
+  useEffect(() => {
+    dataFetch(
+      '/api/provider/capabilities',
+      {
+        method: 'GET',
+        credentials: 'include',
+      },
+      (result) => {
+        if (result) {
+          const capabilitiesRegistry = result;
+          const patternsCatalogueCapability = capabilitiesRegistry?.capabilities.filter(
+            (val) => val.feature === MesheryPatternsCatalog,
+          );
+          if (patternsCatalogueCapability?.length > 0) setCanPublishPattern(true);
+        }
+      },
+      (err) => console.error(err),
+    );
+  }, []);
+
+  // @ts-ignore
+  useEffect(() => {
+    document.body.style.overflowX = 'hidden';
+
+    fetchPatterns(page, pageSize, search, sortOrder, visibilityFilter);
+    return () => (document.body.style.overflowX = 'auto');
+  }, [page, pageSize, search, sortOrder, visibilityFilter]);
+
+  useEffect(() => {
+    if (viewType === 'grid') {
+      setSearch('');
+    }
+  }, [viewType]);
+
+  const initPatternsSubscription = (
+    pageNo = page.toString(),
+    pagesize = pageSize.toString(),
+    searchText = search,
+    order = sortOrder,
+  ) => {
+    if (disposeConfSubscriptionRef.current) {
+      disposeConfSubscriptionRef.current.dispose();
+    }
+    const configurationSubscription = ConfigurationSubscription(
+      (result) => {
+        setPage(result.configuration?.patterns?.page || 0);
+        setPageSize(result.configuration?.patterns?.page_size || 0);
+        setCount(result.configuration?.patterns?.total_count || 0);
+        handleSetPatterns(result.configuration?.patterns?.patterns);
+      },
+      {
+        applicationSelector: {
+          pageSize: pagesize,
+          page: pageNo,
+          search: searchText,
+          order: order,
+        },
+        patternSelector: {
+          pageSize: pagesize,
+          page: pageNo,
+          search: searchText,
+          order: order,
+        },
+        filterSelector: {
+          pageSize: pagesize,
+          page: pageNo,
+          search: searchText,
+          order: order,
+        },
+      },
+    );
+    disposeConfSubscriptionRef.current = configurationSubscription;
+  };
+
+  //needs sistent schema
+  useEffect(() => {
+    dataFetch(
+      '/api/schema/resource/publish',
```

Comment:
this schema is for publish so this is not required here because published items can only be unpublished here